src/scripting/worldmap_sector.cpp

```diff
@@ -59,6 +59,13 @@ WorldMapSector::move_to_spawnpoint(const std::string& spawnpoint)
   m_parent->move_to_spawnpoint(spawnpoint);
 }
 
+const std::string&
```

Comment:
"Mom, can we have `const` correctness?"
- "No, son, we have `const` correctness at home"
Const correctness at home:

Reply:
What @tylerandari13 did, has its purpose, even though it's something I rarely see in this form.

a) Returning a reference is potentially less expensive in terms of memory usage and "effort" (a reference uses less memory than a string afaik) but if the thing we return goes out of scope (not applicable in this case) or is garbage-collected (object gets destroyed) (likely), then we potentially have a problem since the reference points to invalid memory / garbage. 

b) `const correctness` is more of a convention than anything. By adding `const` to the end of the function declaration, you tell the compiler that the function does not modify the state of the object in any way (has no side effects). It's something that most compilers already check when compiling the program and if a class tries to modify its state inside a const method, then the compiler will produce an error.

Sorry, this is awfully technical. If you want, I could maybe break it down for you even further.

Reply:
> a) Returning a reference is potentially less expensive in terms of memory usage and "effort" (a reference uses less memory than a string afaik) but if the thing we return goes out of scope (not applicable in this case) or is garbage-collected destroyed (set to null) (likely at some point), then we potentially have a problem since the reference points to invalid memory / garbage.

In this case, this can only go out-of-scope, when the worldmap is destroyed, which happens when it's exited. If any code needs to preserve this, in my opinion, it should do the copying, and the function should return a `const` reference, so it's less expensive when it doesn't have to be.